backend/src/org/commcare/suite/model/EntityDatum.java

```diff
@@ -119,6 +119,9 @@ public TreeReference getEntityFromID(EvaluationContext ec, String elementId) {
         //Get root nodeset
         TreeReference nodesetRef = this.getNodeset().clone();
         Vector<XPathExpression> predicates = nodesetRef.getPredicate(nodesetRef.size() - 1);
+        if (predicates == null) {
```

Comment:
Not sure, but it looks like the way that `TreeReferenceLevel.getPredicates()` is implemented, it can return `null`, so this handles that case.